source/js/components/join/join.jsx

```diff
@@ -60,6 +60,9 @@ export default class JoinUs extends React.Component {
         source: window.location.toString()
       };
 
+      payload.givenNames = this.givenNames && this.givenNames.value;
```

Comment:
Let's harden this a tiny bit by not even setting payload keys:
```
if (this.givenNames && this.surname) {
  payload.givenNames = this.givenNames.value;
  payload.surname = this.surname.value;
}
```
Mostly because I'm worried that the conditionals can yield `false`, which _may_ end up getting sent to basket, which _may_ end up being interpreted as a string. It probably isn't, but better safe than sorry?

Reply:
I want to leave the option for filling one out but not the other. How's this?

```
this.givenNames? payload=this.givenNames.value : null
this.surname? payload=this.surname.value:null
```